File: src/app/conjugation/forms/causative-passive-negative.ts
import {Conjugation, Word, WordType} from "../conjugation";
import {CausativeAffirmative} from "./causative-affirmative";
import {PassiveNegative} from "./passive-negative";

export class CausativePassiveNegative implements Conjugation {
    getConjugation(word: Word): (Word | undefined) {

        const originalType = word.wordType;
        const causative = new CausativeAffirmative().getConjugation(word);
        if (causative === undefined) {
            return undefined;
        }

        return new PassiveNegative().getConjugation(causative.changeType(WordType.GodanVerb))?.changeType(originalType)
    }

}

File: backend/app/conjugation/forms/causative_passive_negative.py
from ..core import Conjugation, Word, WordType
from .causative_affirmative import CausativeAffirmative
from .passive_negative import PassiveNegative


class CausativePassiveNegative(Conjugation):

    title = 'Causative-Passive, negative'
    settings_title = 'Negative'

    def conjugate(self, word: Word) -> Word | None:
        original_type = word.word_type
        causative = CausativeAffirmative().conjugate(word)
        if causative is None:
            return None

        passive = PassiveNegative().conjugate(
            causative.change_type(WordType.GODAN_VERB)
        )
        if passive is None:
            return None

        return passive.change_type(original_type)
